app/models/code_school.rb

```diff
@@ -2,4 +2,23 @@ class CodeSchool < ApplicationRecord
   validates :name, :url, :logo, presence: true
   validates_inclusion_of :full_time, :hardware_included, :has_online, :online_only, :in => [true, false]
   has_many :locations, -> { order('state ASC, city ASC') }, dependent: :destroy
+
+  before_create :check_scheme
+
+  private
+
+  def check_scheme
+    uri = URI.parse(url)
+    update_url(uri) if uri.scheme.nil?
+  end
```

Comment:
Currently `check_scheme` reads the URL attribute, then `update_url` writes it.

To make the URL handling code reusable the API might look something like this:
```ruby
self.url = ensure_url_protocol(url)
```
For now though, I'd probably keep it all in one method.